common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -98,10 +99,10 @@ public ExternalBlockHandler(
       OneForOneStreamManager streamManager,
       ExternalShuffleBlockResolver blockManager,
       MergedShuffleFileManager mergeManager) {
-    this.metrics = new ShuffleMetrics();
     this.streamManager = streamManager;
     this.blockManager = blockManager;
     this.mergeManager = mergeManager;
+    this.metrics = new ShuffleMetrics();
```

Comment:
Why has this been moved down? Curious.

Reply:
Great question! Changing from:
```
      allMetrics.put("registeredExecutorsSize",
                     (Gauge<Integer>) () -> blockManager.getRegisteredExecutorsSize());
```
to
```
      allMetrics.put("registeredExecutorsSize",
          (Gauge<Integer>) blockManager::getRegisteredExecutorsSize);
```
Means that the `blockManager` object is evaluated at the time of `new ShuffleMetrics()`, instead of later at the time the metrics are gathered. When the instantiation is the first line, `blockManager` is still null. If it's at the end, `blockManager` has been initialized by then.

Reply:
This change was reverted in a subsequent commit. Resolving.